lib/internal/modules/run_main.js

```diff
@@ -59,6 +59,29 @@ function resolveMainPath(main) {
   return mainPath;
 }
 
+// This function fixes a bug where the module loader would load the wrong file
+// (E.G. /app.js instead of /app/index.js with 'node .' in the /app cwd)
+// Adopted from PR #24256
+function maybeAddTrailingSlash(filename) {
+  let trailingSlash = false;
+  if (filename && filename.length) {
+    // We do not want to add a slash to a root path `/`
+    // This is the only path where path.resolve returns a slash at the end
+    trailingSlash = filename.length > 1 &&
+      filename.charAt(filename.length - 1) === path.sep;
+    if (!trailingSlash) {
+      if (path.sep === '/') {
+        // Support *nix /
+        trailingSlash = /(?:^|\/+)\.?\.$/.test(filename);
```

Comment:
This is not how we call Regexes in internal modules afaik. Need to use the Regex primordial for exec.

Reply:
also extract the regexes in their own variables